File: backend/Api/Data/Configurations/Product.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Api.Models;

namespace Api.Data.Configurations
{
    public class ProductConfiguration : IEntityTypeConfiguration<Product>
    {
        public void Configure(EntityTypeBuilder<Product> builder)
        {
            builder.ToTable("Products");
            builder.HasKey(p => p.Id);

            builder.Property(p => p.Name).IsRequired().HasMaxLength(200);
            builder.Property(p => p.SKU).IsRequired().HasMaxLength(100);
            builder.HasIndex(p => p.SKU).IsUnique();
            builder.Property(p => p.Brand).HasMaxLength(100);
            builder.Property(p => p.Description).HasMaxLength(1000);
            builder.Property(p => p.Price).HasColumnType("decimal(18,2)");
            builder.Property(p => p.IsActive).IsRequired().HasDefaultValue(true);
            builder.Property(p => p.CreatedAt).IsRequired();

            builder.HasOne(p => p.Category)
                .WithMany(c => c.Products)
                .HasForeignKey(p => p.CategoryId)
                .OnDelete(DeleteBehavior.SetNull);
        }
    }
}

File: frontend/src/app/core/domain/product/product.ts
import { Category } from '../category/category';

/**
 * Represents a Product
 */
export interface Product {
  /** ID */
  id: number;

  /** Name - product name */
  name: string;

  /** SKU - stock keeping unit */
  sku: string;

  /** Brand - product brand */
  brand?: string | null;

  /** Category ID - FK to Category */
  categoryId?: number | null;

  /** Price - unit price */
  price: number;

  /** Description - product description */
  description?: string | null;

  /** Is Active - is the product active */
  isActive?: boolean;

  /** Category - associated category */
  category?: Category | null;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
